tutorial/source/enumeration.ipynb

```diff
@@ -0,0 +1,440 @@
+    "    print('model x.shape = {}'.format(x.shape))\n",
+    "    print('model y.shape = {}'.format(y.shape))\n",
+    "    print('model z.shape = {}'.format(z.shape))\n",
+    "    \n",
+    "def guide():\n",
+    "    pass\n",
+    "\n",
+    "pyro.clear_param_store()\n",
+    "elbo = TraceEnum_ELBO(max_plate_nesting=0)\n",
+    "elbo.loss(model, guide);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plates and enumeration <a class=\"anchor\" id=\"Plates-and-enumeration\"></a>\n",
+    "\n",
+    "Pyro [plates](http://docs.pyro.ai/en/dev/primitives.html#pyro.plate) express conditional independence among random variables. Pyro's enumeration strategy can take advantage of plates to reduce exponential-cost enumeration of a cartesian product down to a linear-cost enumeration over conditionally independent random variables in lock-step. This is especially important for e.g. minibatched data.\n",
+    "\n",
+    "To illustrate, consider a gaussian mixture model with shared variance and different mean."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Running model with 10 data points\n",
+      "x.shape = torch.Size([10])\n",
+      "Running model with 10 data points\n",
+      "x.shape = torch.Size([3, 1])\n"
+     ]
+    }
+   ],
+   "source": [
+    "@config_enumerate(default=\"parallel\")\n",
+    "def model(data, num_components=3):\n",
+    "    print('Running model with {} data points'.format(len(data)))\n",
+    "    p = pyro.sample(\"p\", dist.Dirichlet(0.5 * torch.ones(3)))\n",
+    "    scale = pyro.sample(\"scale\", dist.LogNormal(0, num_components))\n",
+    "    with pyro.plate(\"components\", num_components):\n",
+    "        loc = pyro.sample(\"loc\", dist.Normal(0, 10))\n",
+    "    with pyro.plate(\"data\", len(data)):\n",
+    "        x = pyro.sample(\"x\", dist.Categorical(p))\n",
+    "        print(\"x.shape = {}\".format(x.shape))\n",
+    "        pyro.sample(\"obs\", dist.Normal(loc[x], scale), obs=data)\n",
+    "        \n",
+    "guide = AutoDiagonalNormal(poutine.block(model, hide=[\"x\", \"data\"]))\n",
+    "\n",
+    "data = torch.randn(10)\n",
+    "        \n",
+    "pyro.clear_param_store()\n",
+    "elbo = TraceEnum_ELBO(max_plate_nesting=1)\n",
+    "elbo.loss(model, guide, data);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Observe that the model is run twice, first by the `AutoDiagonalNormal` to trace sample sites, and second by `elbo` to compute loss. In the first run, `x` has standard interpretation of one sample per datum, hence shape `(10,)`. In the second run enumeration can use the same three values `(3,1)` for all data points, and relies on broadcasting for any dependent sample or observe sites that depend on data."
```

Comment:
has standard interpretation => has the standard interpretation

Reply:
is it worth making "and relies on broadcasting for any dependent sample or observe sites that depend on data" more explicit in the particular context of this model?